core/modules/qhttp/Server.cc

```diff
@@ -129,12 +129,15 @@ void Server::_accept()
     _acceptor.async_accept(
         *socket,
         [self, socket](boost::system::error_code const& ec) {
-            self->_accept(); // start accept for the next incoming connection
+            if (!self->_acceptor.is_open() || ec == asio::error::operation_aborted) {
```

Comment:
Just curious. Do you really need them (tests) both here? The documentation on the method recommends:
```
void accept_handler(const boost::system::error_code& error)
{
  if (!error)
  {
    // Accept succeeded.
  }
}
```
Which, I would assume implies that your may return from the handler if:
```
if (ec == asio::error::operation_aborted) return;
```

Reply:
In practice, both tests are, in fact, needed.  The first condition handles all situations related to acceptor close(); the second situation can and does occur in isolation when a previously closed acceptor is subsequently re-opened (a situation that occurs with the qhttp Server open()/close() protocol, but isn't well explored in the boost documentation, where most examples are single-shot).